### PR TITLE
feat: endpoint to get all events registration status for user

### PR DESCRIPTION
**Pull request by anversa-pro**
## Describe your changes
Added new API endpoint to get the user registration status for all events. 
Implemented validation logic to check the user's registration status and return the result based on that status. 
Implemented tests for both the controller and service of the newly added endpoint
## Issue ticket number and link
https://trello.com/c/vNVLaOq9
## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

### CODE REVIEW

File: services/user.service.js [hunk 1]
```diff
@@ -186,6 +186,40 @@ class UserService extends BaseService {
...
+          id: _id.toString(),
+          name,
+          status,
+          eventDate,
+          description: updatedDescription,
+          subscribed: isUserSubscribed,
+        }
+      }
+    )
+    console.log(updatedEvents)
```
Comment (FelipeLosada1996) [inline services/user.service.js:220]: Remove console.log

File: controllers/v1/users/user.controller.js [hunk 2]
```diff
@@ -77,6 +77,36 @@ const getUserAttendance = async (request, response) => {
...
+      data,
+      message,
+    }
+
+  } catch (err) {
+    return response
+      .status(500)
+      .json(baseController.getErrorResponse('Error verifying user attendance'))
+  }
+  return response.status(200).json(responseData)
```
Comment (jlora-mck) [inline controllers/v1/users/user.controller.js:108]: What about doing this small refactor

```suggestion
  if (!userId) {
    return response.status(400).json(baseController.getErrorResponse('Parameters are missing'))
  }

  try {
    const { status, data, message } = await userService.fetchUserEventsAttendance(
      userId,
      eventService
    )

    return {
      status,
      data,
      message,
    }

  } catch (err) {
    return response
      .status(500)
      .json(baseController.getErrorResponse('Error verifying user attendance'))
  }
```

File: docker-compose.yml [hunk 3]
```diff
@@ -11,17 +11,17 @@ services:
...
+  #   build:
+  #     context: BE-CHUPITO/
+  #     dockerfile: Dockerfile
+  #   image: be-chupito
+  #   ports:
+  #     - "5002:5002"
+  #   depends_on:
+  #     - mongo
+  #   environment:
+  #     - MONGODB_URI=mongodb://mongoDev:Passw0rd@mongo:27017/?authSource=admin
```
Comment (jlora-mck) [inline docker-compose.yml:24]: what are these changes ?

Comment (FelipeLosada1996) [reply docker-compose.yml:24]: remove this change, is not necessary

Comment (anversa-pro) [reply docker-compose.yml:24]: That file should'n load, is the bug reported by Kelly, I comment that to avoid the error an run locally

Comment (anversa-pro) [reply docker-compose.yml:24]: Changes removed

File: controllers/v1/users/user.controller.js (hunk 2 above)
Comment (anversa-pro) [reply controllers/v1/users/user.controller.js]: Applied! 👍🏼

**Review by FelipeLosada1996 — APPROVED** [review]

File: services/user.service.js [hunk 4]
```diff
@@ -186,6 +186,39 @@ class UserService extends BaseService {
...
+    try {
+      const eventsData = await eventsService.doList({})
+      if (!eventsData) {
+        return this.getSuccessResponse({},'No existing events to check')
+      }
+      const responseData = await this.addRegisteredField(userId, eventsData.data)
+
+      return this.getSuccessResponse(responseData, 'Events retrieved successfully.')
+    } catch (err) {
+      return (response = this.getErrorResponse('Error getting user information'))
```
Comment (jlora-mck) [inline services/user.service.js:201]: response is not needed here, please remove in line 190
```suggestion
      return this.getErrorResponse('Error getting user information')
```

File: services/user.service.js [hunk 5]
```diff
@@ -186,6 +186,39 @@ class UserService extends BaseService {
...
+          status,
+          eventDate,
+          description: updatedDescription,
+          subscribed: isUserSubscribed,
+        }
+      }
+    )
+    return updatedEvents
+  }
 }
```
Comment (jlora-mck) [inline services/user.service.js:222]: This method is the depiction of good javascript ❤️

Comment (anversa-pro) [reply services/user.service.js:222]: Thanks! Lessons learned

**Review by jlora-mck — APPROVED** [review]

**Pull request merged**